packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -393,4 +429,50 @@ export class SnapInterfaceController extends BaseController<
       (id: string) => this.messagingSystem.call('SnapController:get', id),
     );
   }
+
+  #_onNotificationsListUpdated(notificationsList: Record<string, any>[]) {
+    const snapNotificationsWithInterface = notificationsList.filter(
+      (notification) => {
+        return notification.type === 'snap' && notification.data?.detailedView;
+      },
+    );
+
+    const interfaceIdSet = new Set(
+      snapNotificationsWithInterface.map(
+        (notification) => notification.data.detailedView.interfaceId,
+      ),
+    );
+
+    const updatedState = Object.entries(this.state.interfaces).reduce<
+      Record<string, StoredInterface>
+    >((newState, [id, snapInterface]) => {
+      if (snapInterface.contentType === ContentType.Notification) {
+        if (interfaceIdSet.has(id)) {
+          newState[id] = snapInterface;
+        }
+      } else {
+        newState[id] = snapInterface;
+      }
+      return newState;
+    }, {});
+
+    this.update((state) => {
+      // @ts-expect-error - TS2589: Type instantiation is excessively deep and
+      // possibly infinite.
+      state.interfaces = updatedState;
+    });
+  }
+
+  /**
+   * Run controller teardown process and unsubscribe from notification service controller events.
+   */
+  destroy() {
+    super.destroy();
+
+    /* eslint-disable @typescript-eslint/unbound-method */
+    this.messagingSystem.unsubscribe(
```

Comment:
Given that I find can't any existing pattern in `core` for this, I think we should just skip unsubscribing for now to reduce complexity.

Reply:
But that's a memory leak though skipping unsubscription no?

Reply:
Technically yes, but with the extension lifecycle I'm not certain it's actually a problem. Especially given that we aren't tearing down most of these elsewhere either